### PR TITLE
Solution for dataStructures/uniqueItems

### PR DESCRIPTION
**Pull request by bensengupta**
Adds a solution for dataStructures/uniqueItems #14

### CODE REVIEW

File: dataStructures/uniqueItems/solutionA/src/main/java/oointro/DungeonLoot.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    HashSet<Loot> storeLoot = new HashSet<>();
+
+    //methods
+    public void addLoot(Loot newLoot){
+        storeLoot.add(newLoot);
+    }
+
+    public void removeLoot(String name){
+        storeLoot.removeIf(loot -> loot.getName() == name);
+    }
```
Comment (judiMc) [inline dataStructures/uniqueItems/solutionA/src/main/java/oointro/DungeonLoot.java:45]: that is a really good use of a lambda expression.  Its going to confuse a lot of  people though.   Could you write a non-lambda version of the method and leave it commented out just below this one with a bit of commentary about the  two?

File: dataStructures/uniqueItems/solutionA/src/main/java/oointro/DungeonLoot.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    public void printSortedLoot(){
+        List<Loot> lootList = new ArrayList<>(storeLoot);
+
+        Collections.sort(lootList, new LootComparator());
+
+        for (Loot l : lootList) {
+            System.out.println(l.toString());
+        }
+    }
+}
```
Comment (judiMc) [inline dataStructures/uniqueItems/solutionA/src/main/java/oointro/DungeonLoot.java:56]: very nice solution.  You've clearly had lots of experience with this kind of stuff before!

File: dataStructures/uniqueItems/solutionA/src/main/java/oointro/DungeonLoot.java (hunk 1 above)
Comment (bensengupta) [reply dataStructures/uniqueItems/solutionA/src/main/java/oointro/DungeonLoot.java:45]: Ok good call 👍

File: dataStructures/uniqueItems/solutionA/src/main/java/oointro/DungeonLoot.java (hunk 2 above)
Comment (bensengupta) [reply dataStructures/uniqueItems/solutionA/src/main/java/oointro/DungeonLoot.java:56]: Thanks! 😄

**Pull request merged**